packages/@react-spectrum/combobox/test/ComboBox.test.js

```diff
@@ -1566,6 +1566,41 @@ describe('ComboBox', function () {
       expect(onSelectionChange).not.toHaveBeenCalled();
     });
 
+    it('retains selected key on tab', function () {
+      let {getByRole} = renderComboBox({allowsCustomValue: true, selectedKey: '2'});
+
+      let combobox = getByRole('combobox');
+      act(() => {
+        userEvent.click(combobox);
+        jest.runAllTimers();
+      });
+
+      act(() => {
+        userEvent.tab();
+        jest.runAllTimers();
+      });
+
+      expect(onSelectionChange).not.toHaveBeenCalled();
+    });
+
+    it('retains selected key on enter', function () {
+      let {getByRole} = renderComboBox({allowsCustomValue: true, selectedKey: '2'});
+
+      let combobox = getByRole('combobox');
+      act(() => {
+        userEvent.click(combobox);
+        jest.runAllTimers();
+      });
+
+      act(() => {
+        fireEvent.keyDown(combobox, {key: 'Enter', code: 13, charCode: 13});
+        fireEvent.keyUp(combobox, {key: 'Enter', code: 13, charCode: 13});
+        jest.runAllTimers();
+      });
+
+      expect(onSelectionChange).not.toHaveBeenCalled();
+    });
```

Comment:
This is a change from prior behavior, but I think this new behavior is more correct. 
Previous behavior:
1. User selects an option from the dropdown.
2. User hits enter on the field
3. onSelectionChange is fired with null to remove the selected key since we figured the user was committing a custom value

New behavior
1. User selects an option from the dropdown.
2. User hits enter on the field
3. Previous selection is not removed since the text matches the selected key text
4. 
I figure the scenario of a user selecting a option and then immediately hitting enter to submit it as a custom value doesn't really happen.